backend/src/test/java/com/speakfluid/backend/entities/ButtonStepTests.java

```diff
@@ -0,0 +1,171 @@
+package com.speakfluid.backend.entities;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import java.util.ArrayList;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+
+/**
+ * Test suite for the ButtonStep entity. Using a variety of Dialogue objects of various suitability for a button choice.
+ * Also incorporates testing for the edge case of an empty Dialogue object.
+ *
+ * @author Aurora Zhang
+ * @version 2.0
+ * @since November 25th, 2022
+ */
+
+
+
+public class ButtonStepTests {
+    ButtonStep button;
+    WozMessage user1;
+    WozMessage chat1;
+    WozMessage user2;
+    WozMessage chat2;
+    WozMessage user3;
+    WozMessage chat3;
+
+    WozMessage chat4;
+
+    WozMessage user4;
+
+    // initialize array to an empty ArrayList otherwise default value is null.
+    ArrayList<WozMessage> userMsgs1 = new ArrayList<>();
```

Comment:
Is there a reason that these have to remain separated? I think this could be better accomplished with an array instead of repeated code